st_ui_builder/lib/ui_builder/type_def.dart

```diff
@@ -40,3 +39,52 @@ typedef GroupComparatorCallback = int Function(
   GroupHeaderData,
   GroupHeaderData,
 );
+
+abstract class StKey {
+  const StKey(this._value);
+
+  final String _value;
+
+  String get value => _value;
+
+  @override
+  String toString() => value;
+
+  @override
+  int get hashCode => value.hashCode;
+
+  @override
+  bool operator ==(Object other) {
+    return other is StKey &&
+        other.runtimeType == runtimeType &&
+        other.value == value;
+  }
+}
+
+class AssetKey extends StKey {
+  const AssetKey(String value) : super(value);
+
+  @override
+  int get hashCode => value.hashCode;
+
+  @override
+  bool operator ==(Object other) {
+    return other is AssetKey &&
+        other.runtimeType == runtimeType &&
+        other.value == value;
+  }
+}
+
+class GameKey extends StKey {
+  const GameKey(String value) : super(value);
+
+  @override
+  int get hashCode => value.hashCode;
+
+  @override
+  bool operator ==(Object other) {
+    return other is GameKey &&
```

Comment:
doesn't "other is GameKey" already give you the required runtimeType comparison??   I'm sleepy but I think it's redundant here ;-)  

But no big deal ... won't hurt anything!!